auto_nag/scripts/severity_underestimated.py

```diff
@@ -0,0 +1,130 @@
+            self.nweeks
+        )
+
+    def has_needinfo(self):
+        return True
+
+    def get_mail_to_auto_ni(self, bug):
+        for field in ["assigned_to", "triage_owner"]:
+            person = bug.get(field, "")
+            if person and not utils.is_no_assignee(person):
+                return {"mail": person, "nickname": bug[f"{field}_detail"]["nick"]}
+
+        return None
+
+    def get_extra_for_needinfo_template(self):
+        return self.extra_ni
+
+    def columns(self):
+        return [
+            "id",
+            "summary",
+            "creation",
+            "last_change",
+            "severity",
+            "dups_count",
+            "votes",
+            "cc_count",
+        ]
+
+    def get_extra_for_template(self):
+        return {
+            "dups_threshold": self.ndups,
+            "votes_threshold": self.votes,
+            "cc_threshold": self.cc,
+        }
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        cc_count = len(bug["cc"])
+        dups_count = len(bug["duplicates"])
+
+        data[bugid] = {
+            "creation": utils.get_human_lag(bug["creation_time"]),
+            "last_change": utils.get_human_lag(bug["last_change_time"]),
+            "severity": bug["severity"],
+            "dups_count": dups_count,
+            "votes": bug["votes"],
+            "cc_count": cc_count,
+        }
+
+        factors = []
+        if dups_count >= self.ndups:
+            factors.append("duplicates")
+        if bug["votes"] >= self.votes:
+            factors.append("votes")
+        if cc_count >= self.cc:
+            factors.append("CCs")
+
+        self.extra_ni[bugid] = {
+            "severity": bug["severity"],
+            "factors": utils.english_list(factors),
+        }
+
+        return bug
+
+    def get_bz_params(self, date):
+        fields = [
+            "assigned_to",
+            "triage_owner",
+            "creation_time",
+            "last_change_time",
+            "severity",
+            "votes",
+            "cc",
+            "duplicates",
+        ]
+
+        params = {
+            "include_fields": fields,
+            "resolution": "---",
+            "bug_severity": ["S3", "S4"],
+            "f1": "keywords",
+            "o1": "nowords",
+            "v1": ["meta", "intermittent"],
+            "f2": "days_elapsed",
+            "o2": "lessthan",
+            "v2": self.nweeks * 7,
+            "j3": "OR",
+            "f3": "OP",
+            "f4": "dupe_count",
+            "o4": "greaterthaneq",
+            "v4": self.ndups,
+            "f5": "votes",
+            "o5": "greaterthaneq",
+            "v5": self.votes,
+            "f6": "cc_count",
+            "o6": "greaterthaneq",
+            "v6": self.cc,
+            "f7": "CP",
+            "n15": 1,
+            "f15": "longdesc",
+            "o15": "casesubstring",
+            "v15": "could you consider increasing the severity?",
```

Comment:
We need to use a different wording or use a different subset of the comment, otherwise this will also exclude the accessibility case from the other script.

Reply:
This was intentional to reduce the noise. If we already asking to increase the severity, then it could be better to not send another request again. Unless we think this will encourage taking an action.

A downside of the current implementation, if the severity was `S4` and increased to `S3` in response to the accessibility case, we will never request increasing the severity again. Also, if the needinfo request got ignored, we will never send another one.

Considering [#1184 (comment)](https://github.com/mozilla/relman-auto-nag/issues/1184#issuecomment-1019921816), instead of excluding bugs in the query level, we could do it by parsing the previous needinfo requests. The `weeks_lookup` threshold that we read have will prevent sending more than one request in a short period of time.

Reply:
Yeah I think we should consider the two things separately. If a bug is S4 and has many duplicates but the triage owner chooses to keep it S4, but then the accessibility team triages it as a access-s2, then we want to push the triage owner to increase the severity.

Ideally we would only post a single comment if both conditions occur (bug is S4 and has many duplicates and access-sX is higher), and we'd only post a new comment when conditions are changed since the last needinfo was answered. I'm not sure how hard it is to implement though, and it's probably not worth it because it should be relatively rare.

Reply:
The needinfo message got reworded in a88252c48d6b78faddd90e2b4936f3a5ea38ba87, and @calixteman's suggestion will be tracked in #1323.